src/ast/passes/semantic_analyser.cpp

```diff
@@ -377,7 +391,8 @@ void SemanticAnalyser::visit(Builtin &builtin)
     builtin.type.SetAS(AddrSpace::kernel);
   }
   else if (builtin.ident == "func") {
-    for (auto &attach_point : *probe_->attach_points)
+    auto probe = dynamic_cast<Probe *>(scope_);
+    for (auto &attach_point : *probe->attach_points)
```

Comment:
I'm guessing these bits of code are the reason for disallowing certain builtins in `check_available_in_subprog()`?

I think it'd be safer if we included the checks closer to the point where they're actually needed (here) - I can imagine in the future we'll forget about this implicit link, modify the list in `check_available_in_subprog` and end up with a SEGV from this dynamic_cast returning a null pointer.

i.e. something like:
```suggestion
    auto *probe = dynamic_cast<Probe *>(scope_);
    if (!probe)
    {
      LOG(ERROR, builtin.loc, err_) << builtin.ident << " cannot be used in functions";
      return;
    }
    for (auto &attach_point : *probe->attach_points)
```